// next-intl/dist/development/navigation/react-client/createSharedPathnamesNavigation.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var ClientLink = require('./ClientLink.js');
var clientRedirect = require('./clientRedirect.js');
var useBasePathname = require('./useBasePathname.js');
var useBaseRouter = require('./useBaseRouter.js');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React__default = /*#__PURE__*/_interopDefault(React);

function createSharedPathnamesNavigation(opts) {
  function Link(props, ref) {
    return /*#__PURE__*/React__default.default.createElement(ClientLink.default, _rollupPluginBabelHelpers.extends({
      ref: ref,
      localePrefix: opts.localePrefix
    }, props));
  }
  const LinkWithRef = /*#__PURE__*/React.forwardRef(Link);
  LinkWithRef.displayName = 'Link';
  function redirect(pathname) {
    for (var _len = arguments.length, args = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
      args[_key - 1] = arguments[_key];
    }
    return clientRedirect.default({
      ...opts,
      pathname
    }, ...args);
  }
  function usePathname() {
    // @ts-expect-error -- Mirror the behavior from Next.js, where `null` is returned when `usePathname` is used outside of Next, but the types indicate that a string is always returned.
    return useBasePathname.default();
  }
  return {
    Link: LinkWithRef,
    redirect,
    usePathname,
    useRouter: useBaseRouter.default
  };
}

exports.default = createSharedPathnamesNavigation;
