// next-intl/dist/development/navigation/react-server/createSharedPathnamesNavigation.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var ServerLink = require('./ServerLink.js');
var serverRedirect = require('./serverRedirect.js');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React__default = /*#__PURE__*/_interopDefault(React);

function createSharedPathnamesNavigation(opts) {
  function notSupported(hookName) {
    return () => {
      throw new Error("`".concat(hookName, "` is not supported in Server Components. You can use this hook if you convert the component to a Client Component."));
    };
  }
  function Link(props) {
    return /*#__PURE__*/React__default.default.createElement(ServerLink.default, _rollupPluginBabelHelpers.extends({
      localePrefix: opts.localePrefix
    }, props));
  }
  function redirect(pathname) {
    for (var _len = arguments.length, args = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
      args[_key - 1] = arguments[_key];
    }
    return serverRedirect.default({
      ...opts,
      pathname
    }, ...args);
  }
  return {
    Link,
    redirect,
    usePathname: notSupported('usePathname'),
    useRouter: notSupported('useRouter')
  };
}

exports.default = createSharedPathnamesNavigation;
