// next-intl/dist/development/navigation/react-server/createLocalizedPathnamesNavigation.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var _rollupPluginBabelHelpers = require('../../_virtual/_rollupPluginBabelHelpers.js');
var React = require('react');
var RequestLocale = require('../../server/react-server/RequestLocale.js');
var utils = require('../shared/utils.js');
var ServerLink = require('./ServerLink.js');
var serverRedirect = require('./serverRedirect.js');

function _interopDefault (e) { return e && e.__esModule ? e : { default: e }; }

var React__default = /*#__PURE__*/_interopDefault(React);

function createLocalizedPathnamesNavigation(_ref) {
  let {
    localePrefix,
    locales,
    pathnames
  } = _ref;
  function Link(_ref2) {
    let {
      href,
      locale,
      ...rest
    } = _ref2;
    const defaultLocale = RequestLocale.getRequestLocale();
    const finalLocale = locale || defaultLocale;
    return /*#__PURE__*/React__default.default.createElement(ServerLink.default, _rollupPluginBabelHelpers.extends({
      href: utils.compileLocalizedPathname({
        locale: finalLocale,
        // @ts-expect-error -- This is ok
        pathname: href,
        // @ts-expect-error -- This is ok
        params: typeof href === 'object' ? href.params : undefined,
        pathnames
      }),
      locale: locale,
      localePrefix: localePrefix
    }, rest));
  }
  function redirect(href) {
    const locale = RequestLocale.getRequestLocale();
    const pathname = getPathname({
      href,
      locale
    });
    for (var _len = arguments.length, args = new Array(_len > 1 ? _len - 1 : 0), _key = 1; _key < _len; _key++) {
      args[_key - 1] = arguments[_key];
    }
    return serverRedirect.default({
      localePrefix,
      pathname
    }, ...args);
  }
  function getPathname(_ref3) {
    let {
      href,
      locale
    } = _ref3;
    return utils.compileLocalizedPathname({
      ...utils.normalizeNameOrNameWithParams(href),
      locale,
      pathnames
    });
  }
  function notSupported(hookName) {
    return () => {
      throw new Error("`".concat(hookName, "` is not supported in Server Components. You can use this hook if you convert the component to a Client Component."));
    };
  }
  return {
    Link,
    redirect,
    getPathname,
    usePathname: notSupported('usePathname'),
    useRouter: notSupported('useRouter')
  };
}

exports.default = createLocalizedPathnamesNavigation;
